spec/controllers/v2/rules_controller_spec.rb

```diff
@@ -97,32 +80,21 @@
   end
 
   describe 'GET show' do
-    let(:rule) { FactoryBot.create(:v2_rule) }
-    let(:security_guide) { rule.security_guide }
+    let(:item) { FactoryBot.create(:v2_rule) }
+    let(:security_guide) { item.security_guide }
 
     context 'Authorized' do
       let(:rbac_allowed?) { true }
 
-      it 'returns rule by id' do
-        item = hash_including('data' => {
-                                'id' => rule.id,
-                                'type' => 'rule',
-                                'attributes' => attributes.each_with_object({}) do |(key, value), obj|
-                                  obj[key.to_s] = rule.send(value)
-                                end
-                              })
-
-        get :show, params: { security_guide_id: security_guide.id, id: rule.id, parents: %i[security_guide] }
-
-        expect(response.parsed_body).to match(item)
+      it_behaves_like 'show', :security_guide do
+        let(:extra_params) { { security_guide_id: security_guide.id, id: item.id } }
       end
 
       context 'under incorrect parent security guide' do
-        let(:item) { FactoryBot.create(:v2_rule) }
         let(:security_guide) { FactoryBot.create(:v2_security_guide) }
 
         it 'returns not_found' do
-          get :show, params: { security_guide_id: security_guide.id, id: rule.id, parents: %i[security_guide] }
+          get :show, params: { security_guide_id: security_guide.id, id: item.id, parents: %i[security_guide] }
 
           expect(response).to have_http_status :not_found
         end
```

Comment:
These seem to be almost the same to me, can we move it also to a shared example into the right (collection/resource) file? Please do the same thing with the rest, ideally we should not have any specs that are not shared, the whole thing was written in a shareable way...